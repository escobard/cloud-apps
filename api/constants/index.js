const dbkeys = require("./dbkeys"),
  routes = require("./routes")

module.exports = {
  dbkeys,
  routes
}
