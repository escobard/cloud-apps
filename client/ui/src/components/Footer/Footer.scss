@import "../../styles/variables";

footer{
  right: 0;
  left: 0;
  bottom: 0;
  border-top: solid 2px $gray-light;
  background: $white;
  text-align: center;
  position: fixed;
  margin: $space-xlrg $space-xlrg 0;
  padding: $space-xxlrg 0 $space-lrg;
  p{
    float: left;
    color: $gray-medium;
    font-weight: 500;
    margin-top: -3px !important;
    span{
      width: $space-lrg;
      display: block;
      float: right;
      margin-left: $space-small;
      background: $gray-medium-dark;
      color: $white;
      border-radius: 50%;
      font-size: $font-size-xs;
    }
  }
  i{
    float: right;
    height: 50px !important;
    background: $blue;
    width: 50px !important;
    border-radius: 50%;
    padding-top: 11.5px;
    color: $white;
    margin-top: -17px !important;
    &:hover{
      cursor: pointer;
    }
  }
}
