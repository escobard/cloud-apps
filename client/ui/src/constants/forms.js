export const addNoteFields = [
  {
    name: "subject",
    label: "Subject",
    placeholder: "Enter a subject for your note",
    errors: ["Subject must contain more than 5"]
  },
  {
    name: "note",
    label: "Note *",
    placeholder: "Enter a description for your note (required)",
    errors: ["Note must contain more than 25 characters"]
  }
];
