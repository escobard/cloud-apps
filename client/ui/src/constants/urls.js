/* istanbul ignore file */

const apiRoot =
  process.env.REACT_APP_DOCKER_VARIANT === "nginx_routing"
    ? "/api"
    : "http://localhost:4000";

export const apiRoutes = {
  addNote: `${`${apiRoot}`}`,
  getNotes: `${`${apiRoot}`}`
};
