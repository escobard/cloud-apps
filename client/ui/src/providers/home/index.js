import HomeProvider, { HomeContext } from "./HomeProvider"

export default HomeProvider;
export {
  HomeContext
}
