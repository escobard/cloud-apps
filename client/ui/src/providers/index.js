import NotesProvider, { NotesContext } from "./notes";

export { NotesProvider, NotesContext };
