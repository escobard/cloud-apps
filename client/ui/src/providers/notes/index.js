import NotesProvider, { NotesContext } from "./NotesProvider";

export default NotesProvider;
export { NotesContext };
