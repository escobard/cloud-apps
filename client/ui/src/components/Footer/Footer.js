import React, { useContext } from "react";

import { useLocation } from "react-router-dom";

import { NotesContext } from "../../providers";

// TODO - fix constant absolute import to allow more readable imports
import { footer } from "../../constants/catalog";

import "./Footer.scss";

const Footer = () => {
  const { openModal, notes } = useContext(NotesContext);
  const { pathname } = useLocation();

  const hasNotes = notes && Array.isArray(notes) && notes.length >= 1;
  const isHome = pathname === "/";

  return (
    <footer>
      {hasNotes && isHome ? (
        <>
          <p>
            {footer.withNotes} <span> {notes.length} </span>
          </p>
        </>
      ) : (
        <p>{footer.noNotes}</p>
      )}
      {notes && Array.isArray(notes) && (
        <i
          aria-label="Add note"
          role="button"
          className="plus big icon"
          onClick={openModal}
          tabIndex={0}
          onKeyDown={(e) => {
            e.key === "Enter" && openModal();
          }}
        />
      )}
    </footer>
  );
};

export default Footer;
