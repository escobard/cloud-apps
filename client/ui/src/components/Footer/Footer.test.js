import { footer } from "constants/catalog";
import { NotesProvider } from "providers";

import Footer from "./index";

describe("Footer", () => {
  const note = {
    title: "Note title",
    note: "Note description"
  };
  const getNotesBody = { data: [note, note, note] };

  afterAll(() => {
    cleanup();
  });

  it(">> snapshot is up to date", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer />{" "}
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it(">> should display no notes when count is 0", () => {
    const { getByText } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer />
      </MemoryRouter>
    );
    expect(getByText(footer.noNotes));
  });

  it(">> should display completed notes and count if count is not 0", async () => {
    mockApi.get.mockResolvedValue(getNotesBody);
    const { getByText } = render(
      <NotesProvider>
        <MemoryRouter initialEntries={["/"]}>
          <Footer />
        </MemoryRouter>
      </NotesProvider>
    );
    await waitForDomChange();
    expect(getByText(footer.withNotes));
  });
});
