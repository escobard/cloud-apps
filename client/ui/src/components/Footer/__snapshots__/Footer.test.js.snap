// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer >> snapshot is up to date 1`] = `
<div>
  <footer>
    <p>
      NO NOTES
    </p>
  </footer>
   
</div>
`;
